packages/fetchai/skills/generic_buyer/strategy.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the strategy class."""
+
+from typing import cast
+
+from aea.helpers.search.models import Constraint, ConstraintType, Description, Query
+from aea.skills.base import SharedClass
+
+DEFAULT_COUNTRY = "UK"
+SEARCH_TERM = "country"
+DEFAULT_MAX_PRICE = 5
+DEFAULT_MAX_BUYER_TX_FEE = 2
+DEFAULT_CURRENCY_PBK = "FET"
+DEFAULT_LEDGER_ID = "fetchai"
+DEFAULT_IS_LEDGER_TX = True
+
+
+class Strategy(SharedClass):
+    """This class defines a strategy for the agent."""
+
+    def __init__(self, **kwargs) -> None:
+        """
+        Initialize the strategy of the agent.
+
+        :return: None
+        """
+        self._country = kwargs.pop("country", DEFAULT_COUNTRY)
```

Comment:
This is specific to a previous implementation, it should made generic..the buyer, like the seller, must specify a datamodel for what it is looking for. See below.